packages/react-native-codegen/src/parsers/parsers-commons.js

```diff
@@ -100,10 +107,73 @@ function emitMixedTypeAnnotation(
   });
 }
 
+function getTypeScriptMemberType(maybeEnumDeclaration: $FlowFixMe) {
+  if (maybeEnumDeclaration.members[0].initializer) {
+    return maybeEnumDeclaration.members[0].initializer.type
+      .replace('NumericLiteral', 'NumberTypeAnnotation')
+      .replace('StringLiteral', 'StringTypeAnnotation');
+  }
+
+  return 'StringTypeAnnotation';
+}
+
+function getFlowMemberType(maybeEnumDeclaration: $FlowFixMe) {
+  return maybeEnumDeclaration.body.type
+    .replace('EnumNumberBody', 'NumberTypeAnnotation')
+    .replace('EnumStringBody', 'StringTypeAnnotation');
+}
+
+function translateDefault(
+  hasteModuleName: string,
+  typeAnnotation: $FlowFixMe,
+  types: TypeDeclarationMap,
+  nullable: boolean,
+  language: ParserType,
```

Comment:
We can remove the `language` parameter if we apply the other changes I'm suggesting.

Reply:
I guess we would still need the `language` parameter for the `UnsupportedEnumDeclarationParserError` call, unless we add some method to the `Parser` interface that returns the `language`

Reply:
That's a good idea. I was thinking the same, honestly. At the end of the day, the Parser implies the language in the type, so it is a good intermediate step to add a `language(): ParserType` property to avoid to pass an extra argument.

Could you take care of that, please? 🙏